PurchasesCoreSwift/SubscriberAttributes/SubscriberAttributesManager.swift

```diff
@@ -0,0 +1,261 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  SubscriberAttributesManager.swift
+//
+//  Created by Madeline Beyl on 8/10/21.
+
+import Foundation
+
+// TODO after migration make class and all methods internal
+@objc(RCSubscriberAttributesManager) public class SubscriberAttributesManager: NSObject {
+
+    private let backend: Backend
+    private let deviceCache: DeviceCache
+    private let attributionFetcher: AttributionFetcher
+    private let attributionDataMigrator: AttributionDataMigrator
+
+    @objc public init(backend: Backend,
+                      deviceCache: DeviceCache,
+                      attributionFetcher: AttributionFetcher,
+                      attributionDataMigrator: AttributionDataMigrator) {
+        self.backend = backend
+        self.deviceCache = deviceCache
+        self.attributionFetcher = attributionFetcher
+        self.attributionDataMigrator = attributionDataMigrator
+    }
+
+    @objc public func setAttributes(_ attributes: [String: String], appUserID: String) {
+        Logger.debug(String(format: Strings.attribution.method_called, "setAttributes"))
```

Comment:
All of these can be replaced with:
`Logger.debug(String(format: Strings.attribution.method_called, #function))`
it's kinda neat. There are a few things like this: https://docs.swift.org/swift-book/ReferenceManual/Expressions.html
![Screen Shot 2021-08-12 at 5 11 00 PM](https://user-images.githubusercontent.com/1304821/129285375-85e87f8b-59b8-4c14-a55f-dd9286733d82.png)

Reply:
oooo neat!! 

i was thinking of ways to de-dupe this code and that would help...was thinking perhaps an Attribute enum that has its key and associated debug string, then we can have one entry point method like `setAttribute(type: AttributeType, value: value, appUserID: appUserID) {
Logger.debug(String(format: type.debugString, #function))
setAttribute(type.key, value, appUserID)
}`

thoughts on doing that post-migration?

Reply:
Post-migration would be 💯